benchmark/Streamly/Benchmark/Data/Fold.hs

```diff
@@ -176,13 +176,34 @@ classifyWith f = S.fold (FL.classifyWith f FL.sum)
 unzip :: Monad m => SerialT m Int -> m (Int, Int)
 unzip = IP.fold $ FL.map (\a -> (a, a)) (FL.unzip FL.sum FL.length)
 
+-------------------------------------------------------------------------------
+-- Stream transformation
+-------------------------------------------------------------------------------
+
+{-# INLINE foldIterateM #-}
+foldIterateM :: Monad m => Int -> SerialT m Int -> m ()
+foldIterateM n =
+    S.drain
+        . S.map getSum
+        . IP.foldIterateM (return . FL.take n . FL.sconcat) (Sum 0)
+        . S.map Sum
+
```

Comment:
Move this to stream benchmarks.

Reply:
A benchmark already existed. Uncommented it.